src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/controller/GenerateRevocationOrderController.kt

```diff
@@ -7,29 +7,74 @@ import org.springframework.http.MediaType
 import org.springframework.http.ResponseEntity
 import org.springframework.security.access.prepost.PreAuthorize
 import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestBody
 import org.springframework.web.bind.annotation.RequestMapping
 import org.springframework.web.bind.annotation.ResponseBody
 import org.springframework.web.bind.annotation.RestController
+import org.thymeleaf.context.Context
+import org.thymeleaf.spring5.SpringTemplateEngine
+import org.thymeleaf.templateresolver.ClassLoaderTemplateResolver
+import uk.gov.justice.digital.hmpps.managerecallsapi.controller.SearchRequest
+import uk.gov.justice.digital.hmpps.managerecallsapi.documents.ClassPathDocumentDetail
+import uk.gov.justice.digital.hmpps.managerecallsapi.documents.HtmlDocumentDetail
 import uk.gov.justice.digital.hmpps.managerecallsapi.documents.PdfDocumentGenerator
+import uk.gov.justice.digital.hmpps.managerecallsapi.search.Prisoner
+import uk.gov.justice.digital.hmpps.managerecallsapi.search.PrisonerOffenderSearchClient
 import java.util.Base64
 
 @RestController
 @RequestMapping(produces = [MediaType.APPLICATION_JSON_VALUE])
 @PreAuthorize("hasRole('ROLE_MANAGE_RECALLS')")
 class GenerateRevocationOrderController(
-  @Autowired private val pdfDocumentGenerator: PdfDocumentGenerator
+  @Autowired private val pdfDocumentGenerator: PdfDocumentGenerator,
+  @Autowired private val prisonerOffenderSearchClient: PrisonerOffenderSearchClient
 ) {
   companion object {
     val log: Logger = LoggerFactory.getLogger(this::class.java)
   }
 
+  private val templateEngine = buildThymeleafEngine()
```

Comment:
It feels like the template engine should be injected with `@Autowired` in the constructor and `buildThymeleafEngine()` (lines 65-74) should be defined in a `@Configuration` class.